src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/RuntimeInteropData.cs

```diff
@@ -61,34 +93,28 @@ public static unsafe int GetStructUnsafeStructSize(RuntimeTypeHandle structureTy
         }
 
         public static IntPtr GetStructUnmarshalStub(RuntimeTypeHandle structureTypeHandle)
-        {
-            if (TryGetStructUnmarshalStub(structureTypeHandle, out IntPtr stub))
+            => TryGetMarshallersForStruct(structureTypeHandle, out _, out IntPtr unmarshal, out _) switch
             {
-                return stub;
-            }
-
-            throw new NotSupportedException(SR.Format(SR.StructMarshalling_MissingInteropData, Type.GetTypeFromHandle(structureTypeHandle)));
-        }
+                InteropDataResult.Invalid => throw new ArgumentException(SR.Format(SR.Arg_CannotMarshal, Type.GetTypeFromHandle(structureTypeHandle)), nameof(structureTypeHandle)),
+                InteropDataResult.NotFound => throw new NotSupportedException(SR.Format(SR.StructMarshalling_MissingInteropData, Type.GetTypeFromHandle(structureTypeHandle))),
+                _ => unmarshal,
+            };
 
         public static IntPtr GetStructMarshalStub(RuntimeTypeHandle structureTypeHandle)
-        {
-            if (TryGetStructMarshalStub(structureTypeHandle, out IntPtr stub))
+            => TryGetMarshallersForStruct(structureTypeHandle, out IntPtr marshal, out _, out _) switch
             {
-                return stub;
-            }
+                InteropDataResult.Invalid => throw new ArgumentException(SR.Format(SR.Arg_CannotMarshal, Type.GetTypeFromHandle(structureTypeHandle)), nameof(structureTypeHandle)),
```

Comment:
```suggestion
                InteropDataResult.Invalid => throw new ArgumentException(SR.Format(SR.Arg_CannotMarshal, Type.GetTypeFromHandle(structureTypeHandle))),
```
The `structureTypeHandle` does match the name of the argument in public signature. It would be better to emit it from the exception.